### ISSUE TITLE
Pre-training 

### ISSUE THREAD

**Issue by kamalkraj** [microsoft/AMOS#3]
Hi,
Any plans on releasing pretraining scripts?
@yumeng5

**Comment by yumeng5** [comment]
Hi,

Although we cannot release the full pretraining script (specifically the `criterion` script for AMOS in fairseq), implementing that should be quite straightforward: The model forward pass for pretraining has already been available in [the model script](https://github.com/microsoft/AMOS/blob/5a7c95786b4fc53bd8c2337ec9ee79b14f5fc0a3/fairseq/fairseq/models/amos/model.py#L146-L176); you only need to implement the generator MLM loss and discriminator binary classification loss in a `criterion` script -- this part will be very similar to COCO-LM (sample code provided in the issue [here](https://github.com/microsoft/COCO-LM/issues/2#issuecomment-1003639940)). I hope this helps!

Best,
Yu
